contracts/ClaimsManager.sol

```diff
@@ -242,35 +239,65 @@ contract ClaimsManager is
         );
     }
 
-    function createClaim(
+    function upgradePolicy(
+        address claimant,
         address beneficiary,
         uint256 coverageAmountInUsd,
         uint256 claimsAllowedFrom,
         uint256 claimsAllowedUntil,
         string calldata policy,
+        string calldata metadata
+    ) external onlyManagerOrPolicyCreator returns (bytes32 policyHash) {
+        policyHash = keccak256(
+            abi.encodePacked(claimant, beneficiary, claimsAllowedFrom, policy)
+        );
+        PolicyState storage policyState = policyHashToState[policyHash];
+        require(policyState.claimsAllowedUntil != 0, "Policy does not exist");
+        require(
+            policyState.coverageAmountInUsd <= coverageAmountInUsd,
+            "Policy coverage amount larger"
+        );
+        require(
+            policyState.claimsAllowedUntil <= claimsAllowedUntil,
+            "Policy allows claims for longer"
+        );
+        policyHashToState[policyHash] = PolicyState({
+            claimsAllowedUntil: uint32(claimsAllowedUntil),
+            coverageAmountInUsd: uint224(coverageAmountInUsd)
```

Comment:
Shouldn't we enforce these types in the args instead of casting?

Reply:
Yes, I'll address that in https://github.com/api3dao/claims-manager/issues/39 because this requires a lot of changes and I didn't want to clutter this PR